templates/user/edit-profile.html

```diff
@@ -269,6 +350,26 @@
                                 </a>
                             </td></tr>
                         {% endif %}
+                        {% if profile.is_totp_enabled %}
+                        <tr>
+                            <td colspan="2"><span class="inline-header grayed">
+                                {{ _('Scratch Codes:') }}
+                                {% if profile.scratch_codes %}
```

Comment:
From what I can tell, the only thing that changes between these two if statement blocks is `Regenerate` vs `Generate`. Why not `if` on that rather than the entire block?